cassandra/cluster.py

```diff
@@ -3964,6 +4000,42 @@ def _refresh_node_list_and_token_map(self, connection, preloaded_results=None,
             log.debug("[control connection] Rebuilding token map due to topology changes")
             self._cluster.metadata.rebuild_token_map(partitioner, token_map)
 
+    def refresh_tablets(self, connection):
+        try:
+            self._refresh_tablets(connection)
+            self._cluster.scheduler.schedule_unique(self._tablet_refresh_time, self._refresh_tablets, connection)
+        except ReferenceError:
+            pass  # our weak reference to the Cluster is no good
+        except Exception:
+            log.debug("[control connection] Error refreshing tablets", exc_info=True)
+            self._signal_error()
```

Comment:
In current implementation, if there is any error while refreshing, it's not going to refresh again ever.
In order to fix that you can move `schedule_unique` after the `try...except` block and put a `return` instead of `pass` in `ReferenceError` case.

Reply:
done